parl/remote/job.py

```diff
@@ -397,25 +400,12 @@ def single_task(self, obj, reply_socket, job_address):
             message = reply_socket.recv_multipart()
             tag = message[0]
             if tag in [
-                    remote_constants.CALL_TAG, remote_constants.GET_ATTRIBUTE,
-                    remote_constants.SET_ATTRIBUTE,
-                    remote_constants.CHECK_ATTRIBUTE
+                    remote_constants.CALL_TAG,
+                    remote_constants.GET_ATTRIBUTE_TAG,
+                    remote_constants.SET_ATTRIBUTE_TAG,
             ]:
                 try:
-                    if tag == remote_constants.CHECK_ATTRIBUTE:
-                        attr = to_str(message[1])
-                        if attr in obj.__dict__:
-                            reply_socket.send_multipart([
-                                remote_constants.NORMAL_TAG,
-                                dumps_return(True)
-                            ])
-                        else:
-                            reply_socket.send_multipart([
-                                remote_constants.NORMAL_TAG,
-                                dumps_return(False)
-                            ])
-
-                    elif tag == remote_constants.CALL_TAG:
+                    if tag == remote_constants.CALL_TAG:
```

Comment:
After calling function, the keys in `__dict__` may be changed. We should also return the latest keys in `__dict__` here.

Reply:
fixed